### PR TITLE
add support for Cortex-R devices

### PR DESCRIPTION
**Pull request by japaric**
r? @alexcrichton 
cc @paoloteti

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1672,6 +1691,9 @@ impl Build {
                         "sparc64-unknown-linux-gnu" => Some("sparc64-linux-gnu"),
                         "sparc64-unknown-netbsd" => Some("sparc64--netbsd"),
                         "sparcv9-sun-solaris" => Some("sparcv9-sun-solaris"),
+                        "armebv7r-none-eabihf" => Some("arm-none-eabi"),
```
Comment (paoloteti) [inline src/lib.rs]: Is this correct? `eb` =  big-endian in `armeb ` . I use 'armeb-eabi' and not `arm-none-eabi` in this case (Linaro toolchain)

Comment (japaric) [reply src/lib.rs]: The `arm-none-eabi-gcc` toolchain can also generate big endian code but it needs the `-mbig-endian` flag. The latest commit addresses that.

``` console
$ arm-none-eabi-readelf -h ./target/armebv7r-none-eabihf/debug/build/foo-728ce1e8d146ec12/out/libfoo.a

File: ./target/armebv7r-none-eabihf/debug/build/foo-728ce1e8d146ec12/out/libfoo.a(foo.o)
ELF Header:
  Magic:   7f 45 4c 46 01 02 01 00 00 00 00 00 00 00 00 00
  Class:                             ELF32
  Data:                              2's complement, big endian
  Version:                           1 (current)
  OS/ABI:                            UNIX - System V
  ABI Version:                       0
  Type:                              REL (Relocatable file)
  Machine:                           ARM
  Version:                           0x1
  Entry point address:               0x0
  Start of program headers:          0 (bytes into file)
  Start of section headers:          1188 (bytes into file)
  Flags:                             0x5000000, Version5 EABI
  Size of this header:               52 (bytes)
  Size of program headers:           0 (bytes)
  Number of program headers:         0
  Size of section headers:           40 (bytes)
  Number of section headers:         22
  Section header string table index: 21
```

``` console
$ arm-none-eabi-readelf -h ./target/armv7r-none-eabi/debug/build/foo-728ce1e8d146ec12/out/libfoo.a

File: ./target/armv7r-none-eabi/debug/build/foo-728ce1e8d146ec12/out/libfoo.a(foo.o)
ELF Header:
  Magic:   7f 45 4c 46 01 01 01 00 00 00 00 00 00 00 00 00
  Class:                             ELF32
  Data:                              2's complement, little endian
  Version:                           1 (current)
  OS/ABI:                            UNIX - System V
  ABI Version:                       0
  Type:                              REL (Relocatable file)
  Machine:                           ARM
  Version:                           0x1
  Entry point address:               0x0
  Start of program headers:          0 (bytes into file)
  Start of section headers:          1168 (bytes into file)
  Flags:                             0x5000000, Version5 EABI
  Size of this header:               52 (bytes)
  Size of program headers:           0 (bytes)
  Number of program headers:         0
  Size of section headers:           40 (bytes)
  Number of section headers:         22
  Section header string table index: 21
```

**Comment by alexcrichton** [comment]
👍

**Comment by japaric** [comment]
@alexcrichton Thanks!

**Pull request merged**